woocommerce/google-listings-and-ads
# Fix that the unsaved prompt should not pop up if the campaign is saved successfully

### Changes proposed in this Pull Request:

Closes #413

- Move API calling of campaign creation and setup finish to an upper level to handle the unsaved checking and API calling in the same place
- Remove unsaved prompt after form submitted
- Remove some files because they're no longer needed for reuse

### Screenshots:

![Kapture 2021-04-01 at 19 39 39](https://user-images.githubusercontent.com/17420811/113289123-62c23d00-9322-11eb-893a-f95b6bf588db.gif)

### Detailed test instructions:

If you want to skip the billing part, you can add a line under the `return <AppSpinner />` condition in [this file](https://github.com/woocommerce/google-listings-and-ads/blob/284997d6a1d712a7b92acfafc081f67b5c1c86fc/js/src/setup-ads/ads-stepper/setup-billing/index.js#L99-L101). For example:
```js
	if ( ! billingStatus ) {
		return <AppSpinner />;
	}

	billingStatus.status = 'approved';
```

1. Delete the gla_ads_setup_completed_at wp option from your database if it exists
    - `delete from wp_options where option_name = 'gla_ads_setup_completed_at';`
2. Head to the Dashboard page: `/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fdashboard`
3. Go to the paid campaign setup page by clicking on the "Add Paid Campaign" button
4. The prompt should not show up when leaving setup without inputting anything on the form. Conversely, if you have entered data, the prompt should pop up when you leave.
5. Finish the setup, and then you should be redirected to the Dashboard page without any interrupted prompt
